### PR TITLE
cups: 2.2.2 -> 2.2.6

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Regular package update but have cupsd use ```Type=notify``` instead of ```Type=simple```.

```nix-build nixos/tests/printing.nix``` passes.

Cc: @jgeerds 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/misc/cups/default.nix [hunk 1]
```diff
@@ -96,7 +101,8 @@ stdenv.mkDerivation rec {
       for f in "$out"/lib/systemd/system/*; do
         substituteInPlace "$f" \
           --replace "org.cups.cupsd" "cups" \
-          --replace "org.cups." ""
+          --replace "org.cups." "" \
+          --replace /var/run /run
```
Comment (vcunat) [inline pkgs/misc/cups/default.nix]: Why this change?  How is the path used?

Comment (peterhoeg) [reply pkgs/misc/cups/default.nix]: ```/var/run``` is deprecated in favour of ```/run```.

Comment (vcunat) [reply pkgs/misc/cups/default.nix]: Ah, right, I remember.  We even have a symlink on NixOS, so it seems unlikely to break anything.

**Comment by orivej** [comment]
I have rebased this on the current staging, cleaned up the original patch, updated `nixos/tests/printing.nix` (now the test passes [1]), and added a patch that fixes the mess with the configure flags. In the process I have discovered that `$lib/lib/libcups.so.2` and `$lib/lib/libcupscgi.so.1` are meant to refer to `$out/share/cups`. Since previously they referred to the literal `$(out)/share/cups` and CUPS was working anyway, I have just nixed those refs. This patch also fixes the CUPS LPD systemd unit which was dysfunctional with its `ExecStart=-$(lib)/lib/cups/daemon/cups-lpd`, although it must have been unused. I have not tested printing on a physical printer yet.

[1] `lpstat -H` is documented to `Show the server hostname and port`, so the old CUPS was wrong to display the socket path. The new CUPS still listens on the socket.

**Comment by orivej** [comment]
WIP: needs more work on Darwin.

**Comment by orivej** [comment]
The `lib/cups` on Linux is `libexec/cups` on Darwin, previously the latter remained in `$lib`, now I move to `$out` just like on Linux.

This is essentially ready, some more testing and I'll merge this.
